aerospike_discovery.py

```diff
@@ -488,54 +575,172 @@ def _authenticate(self, user, password, password_field_id):
 
+use_latencies_cmd = True if int(version[0]) >= 5 and int(version[1]) >= 1 else False
+latency_histogram = ''
+latency_time = ''
+
+if args.dc:
+        asinfo_cmd='dc/'+args.dc
+
+        # Version 5.0+ got removed dc/DC_NAME and added get-stats:context=xdr;dc=DC_NAME
+        if int(version[0]) >= 5:
+            asinfo_cmd='get-stats:context=xdr;dc='+args.dc
+
+# namespace must be checked after sets, bins, and sindex
+elif args.set:
+    asinfo_cmd='sets/'+args.namespace+'/'+args.set
+elif args.bin:
+    asinfo_cmd='bins/'+args.namespace
+elif args.sindex:
+    asinfo_cmd='sindex/'+args.namespace+'/'+args.sindex
+elif args.namespace:
+    asinfo_cmd='namespace/'+args.namespace
+elif args.latency:
+    if use_latencies_cmd:
+        asinfo_cmd='latencies:'
+    else:
+        asinfo_cmd='latency:'
+    if args.stat:
+        latency_histogram, latency_time = args.stat.rsplit('-', 1)
+        print(latency_histogram, latency_time)
+        asinfo_cmd+="hist="+latency_histogram
+
 try:
-    r = client.info(arg_value).strip()
+    res = client.info(asinfo_cmd).strip()
 except Exception as e:
-    print("Failed to execute asinfo command %s on the Aerospike cluster at %s:%s"%(arg_value, args.host, args.port))
-    print e
+    print("Failed to execute asinfo command %s on the Aerospike cluster at %s:%s"%(asinfo_cmd, args.host, args.port))
+    print(e)
     sys.exit(STATE_UNKNOWN)
 
 client.close()
 
-if r == None:
-    print "request to ",args.host,":",args.port," returned no data."
+if res == None:
+    print("request to ",args.host,":",args.port," returned no data.")
     sys.exit(STATE_CRITICAL)
 
-if r == -1:
-    print "request to ",args.host,":",args.port," returned error."
+if res == -1:
+    print("request to ",args.host,":",args.port," returned error.")
     sys.exit(STATE_CRITICAL)
 
-if args.stat != None and args.stat not in r:
-    print "%s is not a known statistic." %args.stat
-    sys.exit(STATE_UNKNOWN)
+# Create lists of latency options
+latency_metrics = ["tps"] 
+if use_latencies_cmd:
+    latency_metrics.extend([str(2**i) + 'ms' for i in range(0, 17)])
+else: 
+    latency_metrics.extend(["1ms", "8ms", "64ms"])
+
+if args.stat is not None:
+    stat = args.stat
+    metric = None
+    if args.latency:
+        # remove appended historgram value.
+        stat, metric = args.stat.rsplit('-', 1)
+    if stat not in res:
+        print("%s is not a known statistic." %args.stat)
+        sys.exit(STATE_UNKNOWN)
+    elif metric and (metric not in latency_metrics):
+        if use_latencies_cmd:
+            print("%s is not a know latencies histogram." % metric)
+        else:
+            print("%s is not a know latency histogram." % metric)
+        sys.exit(STATE_UNKNOWN)
 
-print "{"
-print "\t\"data\":["
-first = True
-r = r.strip()
-for s in r.split(";"):
-    metricname=re.split('=|\t',s)[-2]
-    metricvalue=s.split("=")[-1]
-    if args.stat != None:
-        if args.stat != metricname:
+if res.endswith(";"):
```

Comment:
Please add an example (with input/output) for this.

Reply:
I think you are asking why this is here? I will reference why I added it https://github.com/aerospike/aerospike-collectd/blob/develop/aerospike_plugin.py#L480.  Has to do with some problems that could arise if someone with server pre 3.9 tries to use the plugin.  Should I remove it?

Reply:
Actually i asked to add example string which is being parsed.

Reply:
Would you like one specifically for latency?

Reply:
@sud82 Alright I added examples for each response format